# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/StoragePrincipalHelper.cpp
# Commit: 2fa131e45f54
# Full Hash: 2fa131e45f54cb47e32880bc57ed48a114e5881c
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-07-11 09:13:24
# Description:
#   Bug 1974770 - Add null URI check while producing the partition key for ExpandedPrincipals - r=baku,anti-tracking-reviewers,timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D256823
# ==============================================================================

diff -r 2cc10912f0d1 -r 2fa131e45f54 toolkit/components/antitracking/StoragePrincipalHelper.cpp
--- a/toolkit/components/antitracking/StoragePrincipalHelper.cpp	Thu Jul 10 16:59:17 2025 +0000
+++ b/toolkit/components/antitracking/StoragePrincipalHelper.cpp	Thu Jul 10 17:00:00 2025 +0000
@@ -705,7 +705,7 @@
 
     nsCOMPtr<nsIURI> uri;
     nsresult rv = BasePrincipal::Cast(principal)->GetURI(getter_AddRefs(uri));
-    if (NS_WARN_IF(NS_FAILED(rv))) {
+    if (NS_WARN_IF(NS_FAILED(rv)) || !uri) {
       continue;
     }
 
